### PR TITLE
Feature/python virtualenv

### PR DESCRIPTION
**Pull request by lasote**
Hi! I tried your PR and I detected some failures and I also simplified some other things.
The activate/deactivate files now are `[de]activate_run_python.xxx` and I added a couple of more tests.
Please take a look.

### CODE REVIEW

File: conans/client/generators/virtualenv_python.py [hunk 1]
```diff
@@ -2,12 +2,20 @@
...
     def __init__(self, conanfile):
         super(VirtualEnvPythonGenerator, self).__init__(conanfile)
         self.venv_name = "conanenvpython"
-        self.env["PYTHONPATH"] = list(
-            set(
-                conanfile.env.get("PYTHONPATH", [])
-                + self.deps_env_info.vars.get("PYTHONPATH", [])
-            )
-        )
+        ppath = conanfile.env.get("PYTHONPATH")
```
Comment (birgerbr) [inline conans/client/generators/virtualenv_python.py:9]: When a Conan package does not supply a `PYTHONPATH` the activate script ends up with a `"None"`: `PYTHONPATH="None"${PYTHONPATH+:$PYTHONPATH}`
One woule usually not use `virtualenv_python` when no packages defines a `PYTHONPATH`, but maybe it is better to use `conanfile.env.get("PYTHONPATH", [])`?

File: conans/test/functional/generators/virtualenv_python_test.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
+import platform
+import unittest
+import os
+
+from conans.util.files import load
+
+from conans.test.utils.tools import TestClient
+
+
+class VirtualEnvPythonGeneratorTest(unittest.TestCase):
```
Comment (birgerbr) [inline conans/test/functional/generators/virtualenv_python_test.py:10]: Should we add a test for the `None` case mentioned above?

**Review by birgerbr — COMMENTED** [review]
Thanks for fixing and improving the code :+1:  It workes fine when using on my current projects.
My only comment is about the `"None"` in the `PYTHONPATH` variable.

File: conans/client/generators/virtualenv_python.py (hunk 1 above)
Comment (lasote) [reply conans/client/generators/virtualenv_python.py:9]: Yes, good catch!

File: conans/test/functional/generators/virtualenv_python_test.py (hunk 2 above)
Comment (lasote) [reply conans/test/functional/generators/virtualenv_python_test.py:10]: Sure!

**Comment by lasote** [comment]
Check now please. @birgerbr

**Comment by birgerbr** [comment]
> Check now please. @birgerbr

Nice! LGTM :+1:
Ok if I merge this now?

**Comment by lasote** [comment]
Yes, please
